test/e2e/setup.go

```diff
@@ -57,10 +82,14 @@ func newClientSet() (*clientSet, error) {
 	}
 
 	return &clientSet{
-		OpenshiftClusters: redhatopenshift.NewOpenShiftClustersClient(os.Getenv("AZURE_SUBSCRIPTION_ID"), authorizer),
-		Operations:        redhatopenshift.NewOperationsClient(os.Getenv("AZURE_SUBSCRIPTION_ID"), authorizer),
-		Kubernetes:        cli,
-		MachineAPI:        machineapicli,
+		OpenshiftClusters: redhatopenshift.NewOpenShiftClustersClient(subscriptionID, authorizer),
+		Operations:        redhatopenshift.NewOperationsClient(subscriptionID, authorizer),
+		VirtualMachines:   compute.NewVirtualMachinesClient(subscriptionID, cliAuthorizer),
+		Resources:         features.NewResourcesClient(subscriptionID, cliAuthorizer),
+		ActivityLogs:      insights.NewActivityLogsClient(subscriptionID, cliAuthorizer),
```

Comment:
Tried to reuse `authorizer`, but got various `AuthorizationFailed` errors.

Reply:
@mjudeikis @jim-minter I suggest that we merge with `NewAuthorizerFromCLI` and start benefiting from e2e tests sooner. And after the merge we can get rid of `NewAuthorizerFromCLI` in a separate PR.

If it sounds good - I'll create a separate issue for myself.

Reply:
I think we can do this quite straight forward with something like https://github.com/Azure/ARO-RP/pull/857
It is still missing secret update and creation of SP itself. But maybe the way forward?